README.md

```diff
@@ -67,12 +68,78 @@ iis_site { 'minimal':
 
 ### Types/Providers
 
+* [iis_application](#iis_application)
 * [iis_application_pool](#iis_application_pool)
 * [iis_feature](#iis_feature)
 * [iis_site](#iis_site)
 
 Here, include a complete list of your module's classes, types, providers, facts, along with the parameters for each. Users refer to this section (thus the name "Reference") to find specific details; most users don't read it per se.
```

Comment:
I'm a bit confused with this. This PR is for adding the `iis_application` documenation.